export/mlab_export_test.py

```diff
@@ -0,0 +1,492 @@
+  @mock.patch('mlab_export.rrdtool.fetch')
+  @mock.patch('mlab_export.get_canonical_names')
+  @mock.patch('gzip.open')
+  @mock.patch('StringIO.StringIO.close')
+  def testunit_rrd_export(
+      self, mock_close, mock_open, mock_get_canonical_names, mock_rrdtool_fetch,
+      mock_get_rrd_files):
+    str_fd = StringIO.StringIO()
+    mock_open.return_value = str_fd
+    mock_get_rrd_files.return_value = ['file1']
+    mock_rrdtool_fetch.return_value = [
+        (0, 10, 10), ['value', 'skipped_metric'], [(0.0,0.0)]]
+    mock_get_canonical_names.side_effect = [
+        ('host', 'exp', 'metric'), ('host', 'exp', None)]
+    mock_options = mock.Mock()
+    mock_options.rrddir_prefix = os.path.join(self._testdata_dir, 'rrd')
+    mock_options.ignored_experiments = []
+    mock_options.pretty_json = None
+    mock_options.compress = True
+    mock_options.output = 'output'
+    mock_options.ts_start = 0
+    mock_options.ts_end = 10
+    expected_value = ('{"sample": [{"timestamp": 0, "value": 0.0}], '
+                      '"metric": "metric", "hostname": "host", '
+                      '"experiment": "exp"}\n')
+
+    mlab_export.rrd_export(mock_options)
+    returned_value = str_fd.getvalue()
+
+    self.assertEqual(returned_value, expected_value)
+    mock_get_canonical_names.assert_called_with(
+        'file1', 'skipped_metric', mock_options)
+    mock_open.assert_called_with('output.gz', 'w')
+    self.assertTrue(mock_close.called)
+
+  @mock.patch('mlab_export.default_output_name')
+  @mock.patch('mlab_export.assert_start_and_end_times')
+  @mock.patch('mlab_export.default_start_time')
+  @mock.patch('mlab_export.default_end_time')
+  @mock.patch('mlab_export.logging.basicConfig')
+  @mock.patch('mlab_export.read_metric_map')
+  def testunit_init_args(self, mock_read_metric_map, mock_logging,
+                         mock_default_end_time, mock_default_start_time,
+                         mock_assert_start_and_end_times,
+                         mock_default_output_name):
+    mock_options = disable_show_options(mock.Mock())
+    mock_options.rrddir_prefix = os.path.join(self._testdata_dir, 'rrd')
+    mock_options.exclude = []
+    mock_options.pretty_json = None
+    mock_options.compress = True
+    mock_options.verbose = True
+    mock_options.ts_end = None
+    mock_options.ts_start = None
+    mock_options.update = True
+    mock_options.output = None
+    mock_options.output_dir = '/output'
+    mock_read_metric_map.return_value = {}
+    mlab_export.HOSTNAME = 'mlab2.nuq0t'
+    ts_previous = 0
+
+    mlab_export.init_args(mock_options, ts_previous)
+
+    self.assertTrue(mock_read_metric_map.called)
+    self.assertTrue(mock_logging.called)
+    self.assertTrue(mock_default_start_time.called)
+    self.assertTrue(mock_default_end_time.called)
+    self.assertTrue(mock_assert_start_and_end_times.called)
+    self.assertTrue(mock_default_output_name.called)
+    self.assertEqual(mock_options.rrddir_prefix[-1], '/')
```

Comment:
I feel like we're not getting much assurance out of this test that values are being set correctly. Can we mock the returns and check that they make their way correctly into the `mock_options`?